weblate/vcs/git.py

```diff
@@ -544,57 +544,32 @@ def push(self):
         self.execute(['svn', 'dcommit', self.branch])
 
 
-class GithubRepository(GitRepository):
+class _GitRemoteRepositoryMixin(GitRepository):
```

Comment:
This is not really a mixin, but rather a base class, so I'd prefer cleaner name. How about following?
```suggestion
class GitMergeRequestBase(GitRepository):
```

Reply:
Sounds good, do you need a `_` in the beginning since this is kind of an Internal Implementation detail? :)

Reply:
It's still base class and somebody else might want to build third party vcs backend on that.